File: tech-dhtml/game-connect4/models/Color.js
export class Color {

  static RED = new Color(`red`);
  static YELLOW = new Color(`yellow`);
  #string;

  constructor(string) {
    this.#string = string;
  }

  toString() {
    return this.#string;
  }

  static get(ordinal) {
    return Color.values()[ordinal];
  }

  static values() {
    return [Color.YELLOW, Color.RED];
  }

  static isColorValid(color) {
    return color.toString() === this.YELLOW.toString() || color.toString() === this.RED.toString();
  }
}

File: tech-java-basada-en-objetos/tictactoe/src/main/java/es/usantatecla/tictactoe/Color.java
package es.usantatecla.tictactoe;

enum Color {

	X,
	O,
	NULL;

	private ColorView view = new ColorView();

	public static Color get(int ordinal) {
		assert new ClosedInterval(0, Color.NULL.ordinal()-1).isIncluded(ordinal);
	
 		return Color.values()[ordinal];
	}

	public void write() {
		String string = this.name();
		if (this.isNull()) {
			string = Message.NULL_COLOR.toString();
		}
		this.view.write(string);
	}

	public boolean isNull() {
		return this == Color.NULL;
	}
}
